target_redshift/db_sync.py

```diff
@@ -738,3 +760,48 @@ def sync_table(self):
         else:
             self.logger.info("Table '{}' exists".format(self.schema_name))
             self.update_columns()
+
+    def check_and_return_credentials(self, settings) -> dict[str, Any]:
```

Comment:
What if the target redshift cluster is in a different region? Does this still work?

Reply:
We interact with redshift via connection through pyscopg2 it does not use Boto3

STS is global service so i don't believe it matters what region_name we choose for the client but from past experience it was a required argument.